proposals/lip-0003.md

```diff
@@ -75,35 +78,32 @@ Given the study of the issue presented in the previous section, this LIP propose
 
 The proposed fix is based on hashing each delegate’s unique identifier together with a unique identifier for the round:
 
-- The delegate's unique identifier can be the public key or the ID of the delegate. Arguably, it is better to use the public key since the ID system might be changed in the future. However, this is something to be discussed for the reference implementation. Both options are equal in terms of generating randomness.
-- The round's unique identifier can be the height of the block in which the new round starts or the height of the round itself. As an example, it can be calculated in the same way it is calculated currently, using the height of the round:
+- The delegate's unique identifier is the binary address of the delegate. Note that this proposal does not depend on the implementation of [LIP 0018](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0018.md). The only requirement here is that the addresses have to be unique for each forging delegate which is ensured by the protocol in any case.
+- The round's unique identifier is one of the two random seed generated according to the specifications in [LIP 0022](https://github.com/LiskHQ/lips/blob/master/proposals/lip-0022.md#random-seeds-computation).
 
-```js
-modules.rounds.calc(height).toString();
-```
 
-By having a unique hash per delegate for every round, we will ensure that in every round the list of delegates will be re-ordered based on a uniform distribution.
+By having a unique hash per delegate for every round, we will ensure that in every round the list of delegates will be re-ordered based on a uniform distribution. Moreover, these changes not only fix the uniformity issue (caused by the two issues raised in the [Motivation](#motivation) section), but it will also allow simpler and more human-readable code in any implementation.
 
 ## Specification
 
-The changes will only affect the `Delegates.prototype.generateDelegateList` function mentioned above. These changes consist of removing the nested `for` loops and generating a new list of “delegate hashes” to be re-calculated and ordered every round. In particular, the proposed fix implements a new algorithm that goes as follows:
+The changes affect the `shuffleDelegateListForRound` function mentioned above. These changes consist of removing the nested `for` loops and generating a new list of “delegate hashes” to be re-calculated and ordered every round. In particular, the proposed fix is as follows:
```

Comment:
This is maybe a bit to "implementation" specific. I liked having the reference to the functions in the rationale/motivation but I think the Specifications should not really mention the function name, or the `for` loops. Maybe something more generic: "This section specifies how the delegate set has to be ordered for each round."

Reply:
I agree that it would be better to make this part more timeless and implementation independent.

Reply:
This LIP was one of the first ever written and it is not . Also, the motivation is very specific regarding the _bug_ in the said function. I re-phrased it and simplified it.